storage-provider/server/src/p2p/mod.rs

```diff
@@ -81,56 +57,11 @@ pub(crate) fn deser_keypair<'de, D: de::Deserializer<'de>>(d: D) -> Result<Keypa
     keypair_value_parser(&src).map_err(de::Error::custom)
 }
 
-/// Parses a ED25519 private key into a Keypair.
-/// Takes in a private key or the path to a PEM file, depending on the @ prefix.
-pub(crate) fn keypair_value_parser(src: &str) -> Result<Keypair, String> {
-    let key = if let Some(stripped) = src.strip_prefix('@') {
-        let path = PathBuf::from_str(stripped)
-            .map_err(|e| e.to_string())?
-            .canonicalize()
-            .map_err(|e| e.to_string())?;
-        SigningKey::read_pkcs8_pem_file(path).map_err(|e| e.to_string())?
-    } else {
-        let hex_key = hex::decode(src).map_err(|e| e.to_string())?;
-        SigningKey::try_from(hex_key.as_slice()).map_err(|e| e.to_string())?
-    };
-    Keypair::ed25519_from_bytes(key.to_bytes()).map_err(|e| e.to_string())
-}
-
 /// Parses a string to an optional Peer ID.
 /// Used in the [`ConfigurationArgs`] rendezvous_point field.
-pub(crate) fn string_to_peer_id_option<'de, D: de::Deserializer<'de>>(
-    d: D,
-) -> Result<Option<PeerId>, D::Error> {
-    let s: Option<String> = de::Deserialize::deserialize(d)?;
-    match s {
-        Some(s) => Ok(Some(PeerId::from_str(&s).map_err(de::Error::custom)?)),
-        None => Ok(None),
-    }
-}
-
-/// Runs a bootstrap node from the given config.
-/// The `CancellationToken` is used for a graceful shutdown if the user presses ctrl+c
-pub async fn run_bootstrap_node(
-    config: BootstrapConfig,
-    token: CancellationToken,
-) -> Result<(), P2PError> {
-    tracing::info!("Starting P2P bootstrap node");
-    let (swarm, addr) = config.create_swarm()?;
-
-    tokio::select! {
-        res = bootstrap(swarm, addr) => {
-            if let Err(e) = res {
-                tracing::error!("Failed to start P2P node. Reason: {e}");
-                return Err(e);
-            }
-        },
-        _ = token.cancelled() => {
-            tracing::info!("P2P node has been stopped by the cancellation token...");
-        },
-    }
-
-    Ok(())
+pub(crate) fn string_to_peer_id<'de, D: de::Deserializer<'de>>(d: D) -> Result<PeerId, D::Error> {
```

Comment:
Especially given it is public, I believe this is a better name

```suggestion
pub(crate) fn deserialize_string_to_peer_id<'de, D: de::Deserializer<'de>>(d: D) -> Result<PeerId, D::Error> {
```